src/string.cr

```diff
@@ -3731,6 +3733,36 @@ class String
     self
   end
 
+  private class MatchIterator
```

Comment:
should this be a `struct`, btw?

Reply:
Probably not because it's mutable.

Reply:
A struct can be mutable, but only if not passed around. Otherwise the changes won't be propagated.